### PR TITLE
Ensures external libraries can't be called by callbacks

### PR DESCRIPTION
**Pull request by Y0SH1M4S73R**
## About The Pull Request

It is possible to create a callback whose `object` (the datum it tries to call) is a path to an external library. Needless to say, it's probably a bad idea to allow admins to call arbitrary external libraries. Var-edited callbacks won't be able to reach the point where the library is executed, but only because `WrapAdminProcCall` runtimes when trying to call `CanProcCall` on a string, but if there is some way to call a function that creates a non-varedited callback with an external library as its object, this PR prevents that. 

## Why It's Good For The Game

See above.

## Changelog

:cl:
admin: Admins are unable to invoke functions from external libraries using callbacks.
/:cl:

### CODE REVIEW

File: code/datums/callback.dm [hunk 1]
```diff
@@ -117,6 +117,10 @@
 	if (!object)
 		return
 
+	if(fexists(object))
+		to_chat(usr, "[object] may be an external library. Calling external libraries is disallowed.", confidential = TRUE)
```
Comment (Iamgoofball) [inline code/datums/callback.dm:121]: this should throw a log

Comment (Y0SH1M4S73R) [reply code/datums/callback.dm:121]: I had the exact same thought, but when I asked, Mothblocks said a `to_chat` was good enough.

**Comment by AnturK** [comment]
Not fexists. That will be extremely slow, check if object is text different than GLOBAL_PROC. And put it behind 515 define, since that's whole purpose of splitting call_ext. Honestly not sure this is worth the hassle until 515.

File: code/datums/callback.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/datums/callback.dm:121]: attempted escalation should absolutely be logged, idk what mothblocks is smoking

**Comment by Mothblocks** [comment]
lock behind byond_major <= 515 please so we know to remove this check later on

**Comment by Y0SH1M4S73R** [comment]
> lock behind byond_major <= 515 please so we know to remove this check later on

conditional compile or runtime check?

**Comment by Mothblocks** [comment]
compile check

**Pull request merged**